src/custom_gates.jl

```diff
@@ -18,6 +42,130 @@ function matrix_rep(g::DoubleExcitation)
     return SMatrix{16,16,ComplexF64}(mat)
 end
 
+"""
+    DoubleExcitationPlus(ϕ)
+
+Generate the matrix representation of the [DoubleExcitationPlus](https://docs.pennylane.ai/en/stable/code/api/pennylane.DoubleExcitationPlus.html) gate.
+
+This gate performs an SO(2) rotation in the subspace {|1100⟩, |0011⟩} with a phase-shift on other states:
+
+- |0011⟩ ⟼ cos(φ/2)|0011⟩ - sin(φ/2)|1100⟩
+- |1100⟩ ⟼ cos(φ/2)|1100⟩ + sin(φ/2)|0011⟩
+- |x⟩ ⟼ e^{iφ/2}|x⟩   for all other basis states |x⟩
+
+# Examples
+
+```jldoctest
+julia> ϕ = 3.56;
+julia> gate_matrix = DoubleExcitationPlus(ϕ);
+julia> m  = matrix_rep(gate_matrix);
+julia> eq1 = m * [1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1];
+julia> eq2 = m * [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1];
+julia> eq1 ==  [exp(im*ϕ/2), 0, 0, cos(ϕ/2), 0, 0, 0, 0, 0, 0, 0, 0, sin(ϕ/2), 0, 0, exp(im*ϕ/2)]
+true
+julia> eq2 ==  [exp(im*ϕ/2), 0, 0, -sin(ϕ/2), 0, 0, 0, 0, 0, 0, 0, 0, cos(ϕ/2), 0, 0, exp(im*ϕ/2)]
+true
+```
+
+"""
+struct DoubleExcitationPlus <: AngledGate{1}
+    angle::NTuple{1,Union{Float64,FreeParameter}}
+    DoubleExcitationPlus(angle::T) where {T<:NTuple{1,Union{Float64,FreeParameter}}} =
+        new(angle)
+end
+Braket.chars(::Type{DoubleExcitationPlus}) = "G2+(ang)"
+Braket.qubit_count(::Type{DoubleExcitationPlus}) = 4
+Base.inv(g::DoubleExcitationPlus) = DoubleExcitationPlus(-g.angle[1])
+function matrix_rep(g::DoubleExcitationPlus)
+    cosϕ = cos(g.angle[1] / 2.0)
+    sinϕ = sin(g.angle[1] / 2.0)
+    eiϕ2 = exp(im * g.angle[1] / 2.0)
+    mat = diagm(eiϕ2 * ones(ComplexF64, 16))
+
+    mat[4, :] .= 0
+    mat[:, 4] .= 0
+    mat[13, :] .= 0
+    mat[:, 13] .= 0
+    # Apply phase-shift to states outside rotation subspace
+    mat[4, 4]   = cosϕ
+    mat[13, 13] = cosϕ
+    mat[4, 13]  = -sinϕ
+    mat[13, 4]  = sinϕ
+    return SMatrix{16, 16, ComplexF64}(mat)
+end
+"""
+    DoubleExcitationMinus(ϕ)
+
+Generate the matrix representation of the [DoubleExcitationMinus](https://docs.pennylane.ai/en/stable/code/api/pennylane.DoubleExcitationMinus.html) gate.
+
+This gate performs an SO(2) rotation in the subspace {|1100⟩, |0011⟩} with a phase-shift on other states:
+
```

Comment:
Again with the different angle variables :)

Reply:
sorry, I will use `ϕ `everywhere !

Reply:
Either way, it's just good to be consistent!